integration/tests/basic/helpers/rpc_checks.py

```diff
@@ -201,4 +206,172 @@ def assert_equal_fields(result, comparable_object, comparable_fields, keys_mappi
             l = l.lower()
             r = r.lower()
 
-        assert l.lower() == r.lower(), f"The field '{field}' {l} from response  is not equal to {field} from receipt {r}"
+        assert (
+            l.lower() == r.lower()
+        ), f"The field '{field}' {l} from response  is not equal to {field} from receipt {r}"
+
+
+def count_events(
+    neon_trx_receipt: NeonGetTransactionResult,
+    ignored_events=[NeonEventType.InvalidRevision, NeonEventType.StepReset],
+    is_removed=False,
+):
+    events = neon_trx_receipt.get_all_events(ignore_events=ignored_events, is_removed=is_removed)
+    return Counter([event.neonEventType for event in events])
+
+
+def assert_event_by_type(neon_trx_receipt: NeonGetTransactionResult, event_type):
+    events = neon_trx_receipt.get_all_events(filter_by_type=event_type)
+
+    for event in events:
+        match event_type:
```

Comment:
maybe better to delete match event_type and check all events what neon_trx_receipt has in loop ?

Reply:
I refactored this function a little bit (match is remaining but events are checked in a loop).